Add detailed UV dose formula (#662)
watertap-org/watertap: docs/technical_reference/unit_models/uv_aop.rst

   "Exposure time of UV light", ":math:`t`", "exposure_time", None, ":math:`\text{s}`"
   "Electricity demand of components with phase", ":math:`E_{p, j}`", "electricity_demand_phase_comp", "[t, p, j]", ":math:`\text{W}`"
   "Electricity demand of components", ":math:`E_j`", "electricity_demand_comp", "[t, j]", ":math:`\text{W}`"
   "Electricity demand", ":math:`E`", "electricity_demand", "[t]", ":math:`\text{W}`"
   "Electricity efficiency per log order reduction (EE/O)", ":math:`EE/O_{p, j}`", "electrical_efficiency_phase_comp", "[t, p, j]", ":math:`\text{J/}\text{m}^3`"
   "Lamp efficiency", ":math:`\eta`", "lamp_efficiency", None, None

**Users must provide values for and 'fix' certain variables to solve the model with DOF=0. Thus, users should fix**
    * either 'inactivation_rate' or 'rate_constant',
    * either 'photolysis_rate_constant' or 'reaction_rate_constant',
    * two variables out of 'uv_dose', 'uv_intensity' and 'exposure_time',
    * either 'electricity_demand_phase_comp' or 'electrical_efficiency',
    * and 'lamp_efficiency'.

**However, users may later unfix certain variables for optimization purposes.**

> When setting the `uv_dose_type` configuration option to `UVDoseType.calculated`, there are 7 newly introduced variables must be fixed. This is a net result of 13 degrees of freedom. Newly utilized variables that must be fixed include:

Reviewer: ```suggestion
When setting the `uv_dose_type` configuration option to `UVDoseType.calculated`, there are 7 additional variables that must be fixed. This leads to a total of 13 degrees of freedom. Additional variables that must be fixed include:
```

   "Exposure time of UV light", ":math:`t`", "exposure_time", None, ":math:`\text{s}`"
   "Electricity demand of components with phase", ":math:`E_{p, j}`", "electricity_demand_phase_comp", "[t, p, j]", ":math:`\text{W}`"
   "Electricity demand of components", ":math:`E_j`", "electricity_demand_comp", "[t, j]", ":math:`\text{W}`"
   "Electricity demand", ":math:`E`", "electricity_demand", "[t]", ":math:`\text{W}`"
   "Electricity efficiency per log order reduction (EE/O)", ":math:`EE/O_{p, j}`", "electrical_efficiency_phase_comp", "[t, p, j]", ":math:`\text{J/}\text{m}^3`"
   "Lamp efficiency", ":math:`\eta`", "lamp_efficiency", None, None

**Users must provide values for and 'fix' certain variables to solve the model with DOF=0. Thus, users should fix**
    * either 'inactivation_rate' or 'rate_constant',
    * either 'photolysis_rate_constant' or 'reaction_rate_constant',
    * two variables out of 'uv_dose', 'uv_intensity' and 'exposure_time',
    * either 'electricity_demand_phase_comp' or 'electrical_efficiency',
    * and 'lamp_efficiency'.

**However, users may later unfix certain variables for optimization purposes.**

> When setting the configuration options to calculate the uv dose, the 'uv_dose' variable is no longer specified with either 'uv_intensity' and 'exposure_time' specified.

Reviewer: ```suggestion
When setting the configuration options to calculate the UV dose, the 'uv_dose' variable is no longer specified with either 'uv_intensity' and 'exposure_time' specified.
```

> When setting the configuration options to calculate the UV dose, the 'uv_dose' variable is no longer specified with either 'uv_intensity' and 'exposure_time' specified.

Reviewer: What are the configuration options now? Moreover, I don't think you need to explain what's new vs. what is no longer existent in the model. These are good notes for our team to understand the changes, but not necessarily helpful for users that likely didn't get a chance to see the "old version."

Author: I've clarified the `uv_dose_type` configuration for ease of users to understand.

   "Inlet temperature", ":math:`T`", "temperature", "[t]", ":math:`\text{K}`"
   "Inlet pressure", ":math:`P_{in}`", "pressure", "[t]", ":math:`\text{Pa}`"
   "Outlet pressure", ":math:`P_{out}`", "pressure", "[t]", ":math:`\text{Pa}`"
   "Mass flowrate of components", ":math:`M_{p, j}`", "flow_mass_phase_comp", "[t, p, j]", ":math:`\text{kg/s}`"
   "Inlet volumetric flowrate", ":math:`F_{in}`", "flow_vol", "[t]", ":math:`\text{m}^3\text{/s}`"

**NOTE: Variables for 'temperature', 'pressure', 'flow_mass_phase_comp', and 'flow_vol' come from the associated property package as state variables and are accessed via {port_name}.{state_var_name}**

Aside from the inlet feed state variables (i.e., temperature, pressure, component mass flowrates),
the UV AOP model has at least an additional 6 degrees of freedom that

> the user must specify when `uv_dose_type` configuration option to `UVDoseType.fixed`.

Reviewer: ```suggestion
the user must specify when the `uv_dose_type` configuration option is set to `UVDoseType.fixed`.
```

   "Pseudo-first order rate constant", ":math:`k_0 = I \cdot k`"
   "Pseudo-first order rate constant", ":math:`k_0 = k_d + k_i`"
   "Solvent mass balance", ":math:`M_{\text{H2O},out} = M_{\text{H2O},in}`"
   "Solute mass balance", ":math:`M_{p, j, out} = M_{p, j, in} \cdot \exp(D \cdot k)`"
   "Electricity demand of each component with phase", ":math:`E_{p, j} = EE/O_{p, j} \cdot F_{in} \cdot \log_{10}(M_{p, j, in} / M_{p, j, out}) / \eta`"
   "Electricity demand of each component", ":math:`E_j = \max_p E_{p, j}`"
   "Electricity demand", ":math:`E = \max_j E_j`"
   "UV absorbance", ":math:`UVA = -\log_{10}(UVT)`"

> if ``uv_dose_type`` is set to ``UVDoseType.calculated``:

Reviewer: Should mention `UVDoseType.fixed` for the default set of variables that need to fixed.

Author: I've clarified equations and variables that belong to `UVDoseType.fixed` or `UVDoseType.calculated`

.. csv-table::
   :header: "Description", "Equation"

>    "UV dose", ":math:`D = 10^A \cdot (-\log_{10}(UVT))^(-B \cdot \log_{10}(UVT)) \cdot (\frac{S}{S_0} / F_{in})^C \cdot N_{bank}^D`"

Reviewer: The exponent is messed up for B*log10(UVT). I think you can try wrapping in curly braces:
```suggestion
   "UV dose", ":math:`D = 10^A \cdot (-\log_{10}(UVT))^{(-B \cdot \log_{10}(UVT))} \cdot (\frac{S}{S_0} / F_{in})^C \cdot N_{bank}^D`"
```
